added onInteractionBegan callback to onWheel and onDrag

Alba-C opened the pull request:
While onInteractionEnded() may be more useful in the long run, onInteractionBegan() seems to accomplish our immediate task and looks like an easier solution to implement.

ValentinH commented on src/index.js:
```diff
@@ -171,6 +171,7 @@ class Cropper extends React.Component {
         ? restrictPosition(requestedPosition, this.imageSize, this.state.cropSize, this.props.zoom)
         : requestedPosition
       this.props.onCropChange(newPosition)
```
Why not calling it inside `onDragStart`?

ValentinH commented on src/index.js:
```diff
@@ -24,6 +24,7 @@ class Cropper extends React.Component {
   rafZoomTimeout = null
   state = {
     cropSize: null,
+    onWheelStart: false,
```
I would name it something like `hasWheelJust Started` because current naming made me think it was a callback. 😀

ValentinH commented on src/index.js:
```diff
@@ -206,6 +209,15 @@ class Cropper extends React.Component {
     const newZoom = this.props.zoom - (e.deltaY * this.props.zoomSpeed) / 200
     this.setNewZoom(newZoom, point)
+
+    !this.state.onWheelStart &&
```
Could you please use an if block here for better readability?

ValentinH commented on src/index.js:
```diff
@@ -206,6 +209,15 @@ class Cropper extends React.Component {
+    !this.state.onWheelStart &&
+      this.setState({ onWheelStart: true }, () => this.props.onInteractionStart())
+
+    clearTimeout(this.wheelTimer)
```
Please also clear the timeout in `componentWillUnmount()`.

ValentinH commented on src/index.js:
```diff
@@ -206,6 +209,15 @@ class Cropper extends React.Component {
+    clearTimeout(this.wheelTimer)
+    this.wheelTimer = setTimeout(
+      () => this.setState({ onWheelStart: false }, () => this.props.onInteractionEnd()),
+      250
```
Is this enough? I still have to play with it to forge an opinion, I'm not on the computer now ^^

Alba-C commented:
I took your advice and moved the first onInteractionStart() call from onDrag to onDragStart and didn't need onPinchStart, because it also calls onDragStart.  Inside onWheel I added a state check for onInteractionStart so it doesn't call it too many times and paired that with a timer for onInteractionEnd so it only fires after the user has stopped scrolling the wheel for 250ms.

ValentinH commented:
I've just run the code and 250ms looks good to me too.

One more thing, please remove the `package.lock` file, this project uses Yarn. (You can add it to the .gitignore if you want :) )

Alba-C commented:
All changes complete, let me know if you see anything else.

ValentinH commented:
@all-contributors please add @Alba-C for code 🎉
